opera/genesisstore/fileshash/write_file.go

```diff
@@ -16,41 +18,57 @@ type TmpWriter interface {
 	Drop() error
 }
 
+type tmpWriter struct {
+	TmpWriter
+	h hasher.Hash
+}
+
 type Writer struct {
 	backend io.Writer
 
-	openTmp    func() TmpWriter
-	tmps       []TmpWriter
-	tmpSize    uint64
+	openTmp    func(int) TmpWriter
+	tmps       []tmpWriter
 	tmpReadPos uint64
 
 	size uint64
 
 	pieceSize uint64
 }
 
-func WrapWriter(backend io.Writer, pieceSize, tmpFileLen uint64, openTmp func() TmpWriter) *Writer {
+func WrapWriter(backend io.Writer, pieceSize uint64, openTmp func(int) TmpWriter) *Writer {
 	return &Writer{
 		backend:   backend,
 		openTmp:   openTmp,
 		pieceSize: pieceSize,
-		tmpSize:   tmpFileLen,
 	}
 }
 
 func (w *Writer) writeIntoTmp(p []byte) error {
 	if len(p) == 0 {
 		return nil
 	}
-	if w.size/w.tmpSize >= uint64(len(w.tmps)) {
-		w.tmps = append(w.tmps, w.openTmp())
+	if w.size/w.pieceSize >= uint64(len(w.tmps)) {
```

Comment:
@uprendis can you explain the logic here, please?

Reply:
@cyberbono3 would you mean this change specifically, or `filehash` module in general?